contracts/token/ERC20/extensions/ERC20uRWA.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.26;
+
+import {IERC7943} from "../../../interfaces/IERC7943.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {ERC165, IERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import {ERC20Freezable} from "./ERC20Freezable.sol";
+import {ERC20Restricted} from "./ERC20Restricted.sol";
+
+/**
+ * @dev Extension of {ERC20} according to https://eips.ethereum.org/EIPS/eip-7943[EIP-7943].
+ *
+ * Combines standard ERC-20 functionality with RWA-specific features like user restrictions,
+ * asset freezing, and forced asset transfers.
+ */
+// solhint-disable-next-line contract-name-capwords
+abstract contract uRWA20 is ERC20, ERC165, ERC20Freezable, ERC20Restricted, IERC7943 {
+    /// @inheritdoc ERC20Restricted
+    function isUserAllowed(address user) public view virtual override(IERC7943, ERC20Restricted) returns (bool) {
+        return super.isUserAllowed(user);
+    }
+
+    /// @inheritdoc ERC165
+    function supportsInterface(bytes4 interfaceId) public view virtual override(ERC165, IERC165) returns (bool) {
+        return interfaceId == type(IERC7943).interfaceId || super.supportsInterface(interfaceId);
+    }
+
+    /// @inheritdoc IERC7943
+    function isTransferAllowed(address from, address to, uint256, uint256 amount) public view virtual returns (bool) {
+        return (amount <= available(from) && isUserAllowed(from) && isUserAllowed(to));
+    }
+
+    /// @inheritdoc IERC7943
+    function getFrozen(address user, uint256) public view virtual returns (uint256 amount) {
+        return frozen(user);
+    }
+
+    /// @inheritdoc IERC7943
+    function setFrozen(address user, uint256, uint256 amount) public virtual {
+        require(amount <= balanceOf(user), ERC20InsufficientBalance(user, balanceOf(user), amount));
+        _checkFreezer(user, amount);
```

Comment:
```suggestion
        _checkFreezer(user, amount);
        require(amount <= balanceOf(user), ERC20InsufficientBalance(user, balanceOf(user), amount));
```
Maybe more a style preference, but checking ACL looks to me the first thing to do, like a modifier would do.
https://eips.ethereum.org/EIPS/eip-7943#security-considerations
(Same applies inside `forceTransfer(...)`)